bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/WebSocketConnection.java

```diff
@@ -54,16 +58,16 @@
 public class WebSocketConnection {
     private final Logger logger = LoggerFactory.getLogger(WebSocketConnection.class);
     private final Gson gson = new Gson();
-    WebSocketClient webSocketClient;
-    @Nullable
-    Session session;
-    @Nullable
-    Timer pingTimer;
-    @Nullable
-    Timer pongTimeoutTimer;
-    Listener listener;
-    boolean closed;
-    IWebSocketCommandHandler webSocketCommandHandler;
+    private final WebSocketClient webSocketClient;
+    private final IWebSocketCommandHandler webSocketCommandHandler;
+    private final Listener listener;
+
+    private @Nullable Session session;
+    private @Nullable Timer pingTimer;
+    private @Nullable Timer pongTimeoutTimer;
+    private @Nullable Future<?> sessionFuture;
+
+    private boolean closed;
 
     public WebSocketConnection(String amazonSite, List<HttpCookie> sessionCookies,
             IWebSocketCommandHandler webSocketCommandHandler) throws IOException {
```

Comment:
It always bothered me that we are overriding the default executor used by the WebSocketClient. The default one has a min thread count of 8 and all the threads are non-daemon. I'd prefer if we provided our own thread pool so we can use an existing thread pool for all these connections instead of creating new ones each time.
It would also give us an opportunity to give the threads names that would tie them to this binding.